From e6592f2f4601eb2e7e72ba785a64a9351d87c5d3 Mon Sep 17 00:00:00 2001
From: Werner Koch <wk@gnupg.org>
Date: Mon, 23 Jun 2025 12:16:15 +0200
Subject: [PATCH] gpg: Re-add the revocation reason to the sigclass in a "rev"
 record.

* g10/import.c (get_revocation_reason): Fix setting of r_reason.
--

Note that gpgme has not yet support for parsing the revocation reason.
Due to the split of gpgme it was not tested with gpgmeqt which has its
own parser in its regression tests which kicked it when building
debian packages.

GnuPG-bug-id: 7083
Fixes-commit: 3f825b044b2f1db8773f27a96034c925177fe9f0
---
 NEWS         | 3 +++
 g10/import.c | 7 +++++--
 2 files changed, 8 insertions(+), 2 deletions(-)

--- a/NEWS
+++ b/NEWS
@@ -15,6 +15,9 @@ Noteworthy changes in version 2.5.8 (202
   * dirmngr: Do not require a keyserver for "gpg --fetch-key".
     [T7693]
 
+  * gpg: Add the revocation reason to the sigclass of a "rev" line.
+    Regression in 2.5.7.  [T7073]
+
 
   See-also: gnupg-announce/2025q2/000494.html
   Release-info: https://dev.gnupg.org/T7672
--- a/g10/import.c
+++ b/g10/import.c
@@ -3411,6 +3411,7 @@ get_revocation_reason (PKT_signature *si
   size_t reason_n;
   const byte *reason_p;
   int reason_code = 0;
+  const char *reason_string;
   char *freeme;
 
   if (r_reason)
@@ -3427,9 +3428,11 @@ get_revocation_reason (PKT_signature *si
     {
       reason_code = *reason_p;
       reason_n--; reason_p++;
-      revocation_reason_code_to_str (reason_code, &freeme);
-      if (r_reason)
+      reason_string = revocation_reason_code_to_str (reason_code, &freeme);
+      if (r_reason && freeme)
         *r_reason = freeme;
+      else if (r_reason && reason_string)
+        *r_reason = xstrdup (reason_string);
       else
         xfree (freeme);
 
